File: src/CryptoWallet/app/src/main/java/com/secure/jnet/wallet/util/ByteUtility.java
package com.secure.jnet.wallet.util;

public abstract class ByteUtility {

    public static final String byteArrayToHexString(byte[] data) {
        if (data == null) {
            return "";
        } else {
            StringBuffer hexString = new StringBuffer(data.length * 2);

            for(int i = 0; i < data.length; ++i) {
                int currentByte = data[i] & 255;
                if (currentByte < 16) {
                    hexString.append('0');
                }

                hexString.append(Integer.toHexString(currentByte));
            }

            return hexString.toString().toUpperCase();
        }
    }
}

File: src/CryptoWallet/app/src/main/java/com/secure/jnet/wallet/util/ByteUtility.kt
package com.secure.jnet.wallet.util;

@OptIn(ExperimentalStdlibApi::class)
fun byteArrayToHexString(data: ByteArray?): String =
    data
        ?.map {
            it.toHexString()
        }?.joinToString("")
        ?.toUpperCase()
        ?: ""

fun intToByteArray(vararg elements: Int): ByteArray =
    elements
        .map { it.toByte() }
        .toByteArray()